x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/support/MetadataFlattenedMigrationIntegTests.java

```diff
@@ -97,14 +98,14 @@ public void testMigrationWithConcurrentUpdates() throws Exception {
             });
 
             resetMigration();
-            try {
-                waitForMigrationCompletion();
-            } finally {
-                runUpdateRolesBackground.set(false);
-                executor.shutdown();
-            }
+            waitForMigrationCompletion();
+            runUpdateRolesBackground.set(false);
+            assertAllRolesHaveMetadataFlattened();
+        } finally {
+            // Await all role updates before shutting down
+            roleUpdatesCompleteListener.get();
+            terminate(executor);
```

Comment:
Changed to use `terminate` here because it shuts down and awaits with good timeouts for tests.